### ISSUE TITLE
Bioshock Infinite (8870)

### ISSUE THREAD

**Issue by CSahajdacny** [ValveSoftware/Proton#2700]
# Compatibility Report
- Name of the game with compatibility issues: Bioshock Infinite
- Steam AppID of the game: 8870

## System Information
- GPU:  GTX 970
- Driver/LLVM version:  nvidia 418.56
- Kernel version: 4.4.0-148-generic
- Link to full system information report as [Gist]( https://gist.github.com/CSahajdacny/07a4c2e835804bb7368673bd8035d8f4):
- Proton version:4.2-4

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-8870.log](https://github.com/ValveSoftware/Proton/files/3191620/steam-8870.log)

## Symptoms 
The game doesn't work via forced PROTON.

## Reproduction
*Go to the game tab.
![Captura de pantalla_2019-05-17_10-29-43](https://user-images.githubusercontent.com/45825041/57931410-c5c19c80-788e-11e9-8561-00c70afe049a.png)
and force the game to be played via PROTON.  

*Download...

*And click play...
The game doesn't work...

I also observed some OSX content in the DLC tab...
What are the doing there?
![Captura de pantalla_2019-05-17_10-34-10](https://user-images.githubusercontent.com/45825041/57931743-7e87db80-788f-11e9-8710-31037239eaa5.png)

**Comment by kisak-valve** [comment]
Looks like the Windows release of this game uses CEG (#753).

**Comment by dreamer** [comment]
Are there any problems with playing this game nowadays using native version? I finished this game on Linux and remember it working great out of the box...

**Comment by CSahajdacny** [comment]
> Are there any problems with playing this game nowadays using native version? I finished this game on Linux and remember it working great out of the box...

Yes, but the performance of the OpenGL Linux port is poor compared with the DX11 game on windows 7 with the same hardware.. the gap is huge.

The main problem of Linux ports is that once they are released, they get abandoned by porters.... 
This port must be updated to Vulkan by default, but keeping the Legacy version in the beta tab, like Croteam did with Talos Principle.. 

But if the problem here is CEG... I can't believe that a Valve creation is the biggest obstacle for to enjoy PROTON.

**Comment by ghost** [comment]
I played the linux native port and it was fantastic!

**Comment by AngryPenguinPL** [comment]
> Yes, but the performance of the OpenGL Linux port is poor compared with the DX11 game on windows 7 with the same hardware.. the gap is huge.

Tbh I played Bioshock Infinite on Linux and Windows. Performance was similar. I did not notice any issues in the native version for Linux.
Also as you can see at Phoronix benchmarks Bioshock Infinite working even better on Linux than on Windows. Proof here (first chart): https://www.phoronix.com/scan.php?page=article&item=windows-linux-pascal&num=5

> The main problem of Linux ports is that once they are released, they get abandoned by porters....

It is not a porter company fault but the publisher abandons it. So you should blame publishers not porters.

> This port must be updated to Vulkan by default, but keeping the Legacy version in the beta tab, like Croteam did with Talos Principle..

I do not agree with that. The topic has already been discussed many times. Vulkan is a very good technology, but to remember that it is not a wonderful solution that will solve all problems out of box.
To really exploit Vulkan's potential, we need a well-written game engine, preferably written from scratch under the low-level API.
Sometimes Vulkan did not give any benefits, and sometimes even a decrease in performance. As an example, I will give here Dungeons 3, where devs drop Vulkan and back to OGL because it gives better framerates on this game.

Additionally, Vulcan use in new port makes sense but I do not see a reason to do it for old ports, especially since it works well in OpenGL and sometimes even better than on Windows.
It is not economically viable for the publisher. Nobody will invest in this for the old port.
The more so since the game itself is not even maintained on Windows for a long time. So no make any sense.
But as you see in Crotem example, they still developing this game and thats why they pick up Vulkan.

Also by creating an port for old game from OpenGL to Vulkan, you will cut off many players with old hardware, which can still be played perfectly well, and which GPU does not support Vulkan.
In the case of new games this is not a problem. But if gamers bought the game a year ago and played it on their equipment and later it turns out that the game abandons OpenGL and becomes Vulkan only then it is unfair because they can't play on it anymore. Here is an example - Rust.
Even creating the legacy branch you write about is not fair. Because someone bought it, pay for it. Since the game is still being developed and gets new content and when developers force him to stay with legacy branch - so he received an incomplete product and was cheated in this way.

**Comment by frostworx** [comment]
Just fyi: just tried the steam installation with the gog exe (currently on sale, so was worth the try)
and the game works fine (played unti after the baptism).
It doesn't accept the language I set on steam though and falls back to english (no big deal).
Also copied "Galaxy.dll" and "common.dll" which are probably required by the exe.

**Comment by clauzf** [comment]
Native Linux version is working but with issues, mouse moves faster when moving up and left, if you swing cursor up and down, the cursor drifts upwards slowly, same with left and right.

When using Proton, everything is perfect, aside from the session cannot last long, it freezes after about 15 mins. When the game freezes, I switch to the `tail steam-8870.log -f` terminal, the last reported error is the following lines

```
153668.662:010c:0274:trace:seh:NtQueryInformationThread (0x89c,0,0x290ffb44,1c,(nil))
err:   DxvkSubmissionQueue: Command submission failed: VK_ERROR_DEVICE_LOST
153669.073:010c:0278:trace:seh:NtQueryInformationThread (0x8d4,0,0x295dfb14,1c,(nil))
err:   DxvkSubmissionQueue: Failed to sync fence: VK_ERROR_DEVICE_LOST
```

Here is the full Proton logs
[steam-8870.log](https://github.com/ValveSoftware/Proton/files/7474760/steam-8870.log)

**Comment by kisak-valve** [comment]
BIOSHOCK: Infinite

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5395.
@username-murphy posted on 2021-12-10T01:22:33:

# Whitelist Request
- Name of the game to be whitelisted: BIOSHOCK: Infinite
- Steam AppID of the game: BioShock Infinite · AppID: 8870

## System Information
- GPU:nVid
- Driver/LLVM version: 
- Distro version: kDE Neon 20.04 w/ Kwin X11 Compositor [ ON ]
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/username-murphy/cecc0bf365612034a71fc2ccc8aa1484
- Proton version: 6.3-8/ 101221

## I confirm:
- [ X ] that pressing the `Play` button in the Steam client is sufficient.
- [ X ] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options) are NOT necessary to run the game.
- [X ] that no workarounds other than PROTON_* env variables are necessary.

## Issues
- [X ] I haven't experienced any issues.
- [X] There are no issues left open for this game.

**Comment by ghost** [comment]
BIOSHOCK: Infinite

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5395.
@username-murphy posted on 2021-12-10T01:22:33:

# Whitelist Request

Name of the game to be whitelisted: BIOSHOCK: Infinite
Steam AppID of the game:

BioShock Infinite · AppID: 8870

System Information:
GPU: nVidia GTX 1050 2GvRAM
Driver/LLVM version:
nVidia-495 Canonical Launchpad Graphics PPA
Other PPA: 
Kisak PPA / Canonical-X Staging PPA
Kernel:
5.13-23.23-Generic Canonical Proposed w/ nVidia Modules/ Objects/ Signatures
Distro version:
kDE Neon 20.04 w/ Backports and Proposed [ enabled] w/ Kwin X11 Compositor [ ON ]

System Information:
https://gist.github.com/username-murphy/cecc0bf365612034a71fc2ccc8aa1484

Valve Steam Client - Beta / Valve Steam Linux Runtime / Sniper / Soldier -all- [ beta client ]
Proton version: 6.3-8_101221

I confirm:
[ X ] that pressing the Play button in the Steam client is sufficient.
[ X ] that runtime config options are NOT necessary to run the game.
[X ] that no workarounds other than PROTON_* env variables are necessary.
BioShock_Infinite_log
[ steam-8870_BioShock_Infinite.tar.gz](https://github.com/ValveSoftware/Proton/files/7691113/steam-8870_BioShock_Infinite.tar.gz)

[ NO_PROTON_LAUNCH_OPTIONS_USED ]

## Issues
- [X ] I haven't experienced any issues.
- [X] There are no issues left open for this game.
- 
@kisak-valve please note log added.

**Comment by noammor** [comment]
Bioshock Infinite worked perfectly in Proton two days ago, except for a crash once every several hours. Now it gained a launcher program, which doesn't work right in Proton.

In the two times the launcher did work, I was able to launch the game. Now the launcher doesn't run pretty consistently.

Native also doesn't work for me now.

Other games do work, of course (e.g Bioshock 2). I'm using Ubuntu 20.04, Steam beta branch, nvidia drivers version 515.48.07.

**Comment by noammor** [comment]
Workaround:

Paste this into the launch options:

`eval $( echo "%command%" | sed "s/2KLauncher\/LauncherPatcher.exe'.*/Binaries\/Win32\/BioShockInfinite.exe'/" )`

**Comment by csc-chicken** [comment]
pasting this command not work for me .Meanwhile Symlink Bioshock Infinite.exe to LauncherPatcher.exe will have engine .ini missing error. It's annoying. It only worked once when I run it for the first time ever without command by the way

**Comment by Newbytee** [comment]
I get this error when I look into the telescope right after the "Go to the monument and find the girl" message appears. It's rather early in the game. Happens with both Proton 7.0-4 and Experimental.

![image](https://user-images.githubusercontent.com/4428753/202557126-5823dbd4-54e2-470f-a45b-4c50dc35d142.png)
